### PR TITLE
Sass 2506 - Lifetime Allowance View Model and encryption

### PR DESCRIPTION
**Pull request by HarryGull**
### Description
Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[SASS-2506](https://jira.tools.tax.service.gov.uk/browse/SASS-2506)

##### Before Reviewing
- [ ]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [ ]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/models/pension/charges/LifetimeAllowance.scala [hunk 1]
```diff
@@ -16,16 +16,34 @@
...
+import utils.{EncryptedValue, SecureGCMCipher}
+import utils.DecryptableSyntax.DecryptableOps
+import utils.DecryptorInstances.bigDecimalDecryptor
+import utils.EncryptableSyntax.EncryptableOps
+import utils.EncryptorInstances.bigDecimalEncryptor
 
-case class LifetimeAllowance(amount: BigDecimal, taxPaid: BigDecimal)
+case class LifetimeAllowance(amount: BigDecimal, taxPaid: BigDecimal) {
+
+  def encrypted()(implicit secureGCMCipher: SecureGCMCipher, textAndKey: TextAndKey): EncryptedLifetimeAllowance =
```
Comment (abiodxn) [inline app/models/pension/charges/LifetimeAllowance.scala:29]: does this need to be implicit ?

File: app/models/pension/charges/LifetimeAllowance.scala [hunk 2]
```diff
@@ -16,16 +16,34 @@
...
+}
 
 object LifetimeAllowance {
   implicit val format: OFormat[LifetimeAllowance] = Json.format[LifetimeAllowance]
 }
 
-case class EncryptedLifetimeAllowance(amount: EncryptedValue, taxPaid: EncryptedValue)
+case class EncryptedLifetimeAllowance(amount: EncryptedValue, taxPaid: EncryptedValue) {
+
+  def decrypted()(implicit secureGCMCipher: SecureGCMCipher, textAndKey: TextAndKey): LifetimeAllowance = LifetimeAllowance(
```
Comment (abiodxn) [inline app/models/pension/charges/LifetimeAllowance.scala:42]: same here ?

**Review by abiodxn — COMMENTED** [review]
Delete empty file `f.sc` 
Other than that just a couple of comments/suggestions

File: app/controllers/pensions/paymentsIntoPension/PaymentsIntoPensionsCYAController.scala [hunk 3]
```diff
@@ -113,9 +113,28 @@ class PaymentsIntoPensionsCYAController @Inject()(implicit val cc: MessagesContr
...
         prior.pensionCharges.flatMap(a => a.pensionContributions).map(_.pensionSchemeTaxReference)
+      ),
+
+      //TODO: validate and amend if necessary when building the lifetime allowance CYA page
+      pensionLifetimeAllowances = PensionLifetimeAllowancesViewModel(
+        aboveLifetimeAllowanceQuestion = if(prior.pensionCharges.flatMap(a => a.pensionSavingsTaxCharges).map(
+          _.benefitInExcessOfLifetimeAllowance).isDefined || prior.pensionCharges.flatMap(
+          a => a.pensionSavingsTaxCharges).map(_.lumpSumBenefitTakenInExcessOfLifetimeAllowance).isDefined){
+          Some(true)
+        } else { None },
```
[the remark this answers is not in the document]
Comment (HarryGull) [reply app/controllers/pensions/paymentsIntoPension/PaymentsIntoPensionsCYAController.scala]: ok

File: app/models/pension/charges/LifetimeAllowance.scala (hunk 1 above)
Comment (HarryGull) [reply app/models/pension/charges/LifetimeAllowance.scala:29]: yes won't work else - how new solution was designed

File: app/models/pension/charges/LifetimeAllowance.scala (hunk 2 above)
Comment (HarryGull) [reply app/models/pension/charges/LifetimeAllowance.scala:42]: yes won't work else - how new solution was designed

**Review by abiodxn — APPROVED** [review]
LGTM

**Pull request merged**